libraries/chain/webassembly/runtimes/eos-vm.cpp

```diff
@@ -105,6 +105,17 @@ void validate( const bytes& code, const wasm_config& cfg, const whitelisted_intr
       EOS_ASSERT(apply_idx < std::numeric_limits<uint32_t>::max(), wasm_serialization_error, "apply not exported");
       const vm::func_type& apply_type = bkend.get_module().get_function_type(apply_idx);
       EOS_ASSERT((apply_type == vm::host_function{{vm::i64, vm::i64, vm::i64}, {}}), wasm_serialization_error, "apply has wrong type");
+
+      // Check sync-call entry point after the protocol feature is activated.
+      // It is optional for a contract to provide sync calls.
+      // To prevent problems, if sync call entry point is present, its signature must be correct
+      if (control.is_builtin_activated(builtin_protocol_feature_t::sync_call)) {
+         uint32_t sync_call_idx = bkend.get_module().get_exported_function("sync_call");
+         if (sync_call_idx < std::numeric_limits<uint32_t>::max()) {
+            const vm::func_type& sync_call_type = bkend.get_module().get_function_type(sync_call_idx);
+            EOS_ASSERT((sync_call_type == vm::host_function{{vm::i64, vm::i64, vm::i32}, {}}), wasm_serialization_error, "sync call entry point has wrong type");
```

Comment:
This is checking signature right? Maybe "sync call entry point has wrong function signature"

Reply:
https://github.com/AntelopeIO/spring/pull/1266/commits/d84de41da5ca3bb78fd17d35b95f6d608885c1e7